### PR TITLE
New resource/Data source `azurerm_data_share_account`

### PR DESCRIPTION
**Pull request by yupwei68**
fix terraform-providers/terraform-provider-azurerm#6480

datashare account

### CODE REVIEW

File: azurerm/internal/services/datashare/data_source_data_share_account.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[INFO] DataShare %q does not exist - removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("retrieving DataShare Account %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	if resp.ID == nil || *resp.ID == "" {
+		return fmt.Errorf("reading DataShare Account %q (Resource Group %q): ID is empty", name, resourceGroup)
```
Comment (katbyte) [inline azurerm/internal/services/datashare/data_source_data_share_account.go:79]: should this be
```suggestion
		return fmt.Errorf("reading DataShare Account %q (Resource Group %q): ID is empty or nil", name, resourceGroup)
```

File: azurerm/internal/services/datashare/resource_arm_data_share_account.go [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+						},
+					},
+				},
+			},
+
+			// the api will save and return the tag keys in lowercase, so an extra validation of the key is all in lowercase is added
+			"tags": {
+				Type:         schema.TypeMap,
+				Optional:     true,
+				ValidateFunc: validate.DatashareTags,
```
Comment (katbyte) [inline azurerm/internal/services/datashare/resource_arm_data_share_account.go:83]: I think this is also done else where? maybe we want to add a `tags.SchemaEnforceLowerCaseKeys`

File: azurerm/internal/services/datashare/resource_arm_data_share_account.go [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+						},
+						"tenant_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+
+			// the api will save and return the tag keys in lowercase, so an extra validation of the key is all in lowercase is added
```
Comment (katbyte) [inline azurerm/internal/services/datashare/resource_arm_data_share_account.go:79]: could we open an issue on the sdk and link it here?

File: azurerm/internal/services/datashare/tests/resource_arm_data_share_account_test.go [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMDataShareAccountExists(data.ResourceName),
+				),
+			},
+			data.RequiresImportErrorStep(testAccAzureRMDataShareAccount_requiresImport),
+		},
+	})
+}
+
+func TestAccAzureRMDataShareAccount_complete(t *testing.T) {
```
Comment (katbyte) [inline azurerm/internal/services/datashare/tests/resource_arm_data_share_account_test.go:54]: The complete test should just have the complete step. a separate update test should go from basic->complete->basic, or basic->complete->complete-updated->basic

File: azurerm/internal/services/datashare/tests/resource_arm_data_share_account_test.go [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+`, data.RandomInteger, data.Locations.Primary)
+}
+
+func testAccAzureRMDataShareAccount_basic(data acceptance.TestData) string {
+	template := testAccAzureRMDataShareAccount_template(data)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_data_share_account" "test" {
+  name                = "acctest-dsa-%d"
```
Comment (katbyte) [inline azurerm/internal/services/datashare/tests/resource_arm_data_share_account_test.go:147]: can we put some uppercase letters in this name? if possible?

File: website/docs/d/data_share_account.html.markdown [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+* `resource_group_name` - (Required) The name of the Resource Group where the Data Share Account exists.
+
+## Attributes Reference
+
+In addition to the Arguments listed above - the following Attributes are exported: 
+
+* `id` - The ID of the Data Share Account.
+
+* `identity` - A `identity` block as defined below.
```
Comment (katbyte) [inline website/docs/d/data_share_account.html.markdown:44]:
```suggestion
* `identity` - An `identity` block as defined below.
```

**Comment by yupwei68** [comment]
Hi @katbyte Sure! Corresponding changes are pushed

**Pull request merged**